server/neptune/workflows/internal/revision_setter/workflow.go

```diff
@@ -0,0 +1,176 @@
+	var resp activities.ListPRsResponse
+	err := workflow.ExecuteActivity(ctx, r.GithubActivities.ListPRs, activities.ListPRsRequest{
+		Repo:  repo,
+		State: github.Open,
+	}).Get(ctx, &resp)
+	if err != nil {
+		return []github.PullRequest{}, errors.Wrap(err, "listing open PRs")
+	}
+
+	return resp.PullRequests, nil
+}
+
+func (r *Runner) setMinRevisionForPRsModifiyingRoot(ctx workflow.Context, req Request, prs []github.PullRequest) error {
+	// spawn activities to list modified files in each open PR async
+	futuresByPullNum := map[github.PullRequest]workflow.Future{}
+	for _, pr := range prs {
+		futuresByPullNum[pr] = workflow.ExecuteActivity(ctx, r.GithubActivities.ListModifiedFiles, activities.ListModifiedFilesRequest{
+			Repo:        req.Repo,
+			PullRequest: pr,
+		})
+	}
+
+	// resolve the futures and spawn activities to set minimum revision for PR if needed
+	futures := []workflow.Future{}
+	for _, pr := range prs {
+		if setMinimumRevisionFuture := r.setMinimumRevisionForPR(ctx, req, pr, futuresByPullNum[pr]); setMinimumRevisionFuture != nil {
+			futures = append(futures, workflow.ExecuteActivity(ctx, r.RevisionSetterActivities.SetPRRevision, activities.SetPRRevisionRequest{
+				Repository:  req.Repo,
+				PullRequest: pr,
+			}))
+		}
+	}
+
+	// wait to resolve futures for setting minimum revision
+	for _, future := range futures {
+		var resp activities.SetPRRevisionResponse
+		err := future.Get(ctx, &resp)
+		if err != nil {
+			return errors.Wrap(err, "error setting pr revision")
+		}
+	}
+
+	return nil
+}
+
+func (r *Runner) setMinimumRevisionForPR(ctx workflow.Context, req Request, pull github.PullRequest, future workflow.Future) workflow.Future {
+	setMinRevisionFn := func() workflow.Future {
+		return workflow.ExecuteActivity(ctx, r.RevisionSetterActivities.SetPRRevision, activities.SetPRRevisionRequest{
+			Repository:  req.Repo,
+			PullRequest: pull,
+		})
+	}
+
+	// let's be preventive and set minimum revision for this PR if this listModifiedFiles fails after 3 attempts
+	var result activities.ListModifiedFilesResponse
+	if err := future.Get(ctx, &result); err != nil {
+		logger.Error(ctx, "error listing modified files in PR", key.ErrKey, err, key.PullNumKey, pull.Number)
+		return setMinRevisionFn()
+	}
+
+	// should not fail unless the TrackedFiles config is invalid which is validated on startup
+	// let's be preventive and set minimum revision for this PR if file path match errors out and alarm on this
+	rootModified, err := doesPRModifyRoot(req.Root, result.FilePaths)
+	if err != nil {
+		r.Scope.Counter("filepath_match_err").Inc(1)
+		logger.Error(ctx, "error matching file paths in PR", key.ErrKey, err, key.PullNumKey, pull.Number)
+		return setMinRevisionFn()
+	}
+
+	if rootModified {
+		return setMinRevisionFn()
+	}
+
+	return nil
+}
+
+func doesPRModifyRoot(root terraform.Root, modifiedFiles []string) (bool, error) {
```

Comment:
typically "does" is not used.  Also PR isn't even passed in as an arg here. maybe `isRootModified` would be a better name